numba/cuda/cudadrv/devicearray.py

```diff
@@ -354,10 +354,21 @@ def view(self, dtype):
         copy of the data.
         """
         dtype = np.dtype(dtype)
-        if dtype.itemsize != self.dtype.itemsize:
-            raise TypeError("new dtype itemsize doesn't match")
+
+        shape = list(shape)
+        shape[-1], rem = divmod(
+            dtype.itemsize,
+            shape[-1] * self.dtype.itemsize
+        )
+        shape = tuple(shape)
+
+        if rem != 0:
```

Comment:
Do you also need a contiguity check here, to prevent taking views of non-contiguous slices? (I noticed that NumPy doesn't seem to allow reinterpreting an array with a dtype of a different size for non-contiguous arrays)

Reply:
Good point! Yeah I think you are right. Will update.

Reply:
Added. Thanks!